metrics/lib/common.bash

```diff
@@ -95,18 +95,40 @@ check_images()
 # This function performs a docker build on the image names
 # passed in, to ensure that we have the latest changes from
 # the dockerfiles
+build_dockerfile_image()
+{
+	local image="$1"
+	local dockerfile_path="$2"
+
+	echo "docker building $image"
+	if ! docker build --label "$image" --tag "${image}" -< "$dockerfile_path"; then
+		die "Failed to docker build image $image"
+	fi
+}
+
+# This function verifies that the dockerfile version is
+# equal to the test version in order to build the image or
+# just run the test
 check_dockerfiles_images()
 {
 	local image="$1"
 	local dockerfile_path="$2"
+
 	if [ -z "$image" ] || [ -z "$dockerfile_path" ]; then
 		die "Missing image or dockerfile path variable"
-		exit 1;
 	fi
-	echo "docker building $image"
-	if ! docker build --label "$image" --tag "${image}:latest" -< "$dockerfile_path"; then
-		die "Failed to docker build image $image"
-		exit 1;
+
+	# Verify that dockerfile version is equal to test version
+	check_image=$(docker images "$image" -q)
+	if [ -n "$check_image" ]; then
+		check_image_version=$(docker image inspect $image | grep -w DOCKERFILE_VERSION | head -1 | cut -d '"' -f4)
+		dockerfile_version=$(grep DOCKERFILE_VERSION $dockerfile_path | cut -d '"' -f2)
+		if [ "$dockerfile_version" -ne "$check_image_version" ]; then
+			echo "$dockerfile_version is not equal to $check_image_version"
+			build_dockerfile_image "$image" "$dockerfile_path"
```

Comment:
and... if you are doing any edits, it occurs to me that if you refactor the tests, then you could pull this build_dockerfile line out to the bottom and only have one call to it (rather than the two we have right now). So, make the tests check if a build is needed, and return if not - and then finally drop through to a single build line at the bottom of the func if it is needed - iyswim?